src/Distribution/Server/Util/CabalRevisions.hs

```diff
@@ -227,28 +237,78 @@ checkFlag flagOld flagNew = do
 checkPackageDescriptions :: Check PackageDescription
 checkPackageDescriptions
   pdA@(PackageDescription
-     packageIdA licenseA licenseFileA
-     copyrightA maintainerA authorA stabilityA testedWithA homepageA
-     pkgUrlA bugReportsA sourceReposA synopsisA descriptionA
-     categoryA customFieldsA _buildDependsA _specVersionRawA buildTypeA
-     customSetupA _libraryA _subLibrariesA _executablesA _foreignLibsA
-     _testSuitesA _benchmarksA
-     dataFilesA dataDirA extraSrcFilesA extraTmpFilesA extraDocFilesA)
+     { specVersionRaw  = _specVersionRawA
```

Comment:
Hrm...if we do this, how do we detect whether we've deconstructed *all* fields of `PackageDescription` ?

Reply:
(for future work, maybe `Generics` or TH could help us here?)

Reply:
Pick either:
- busywork to sort the fields when we move them around (`buildDepends` will be gone, we know that already)
- busywork to compare that we have all fields.

I think latter is less work

Reply:
Note: what we can do today, is copy the fields which can change, and compare for equality. That won't get us good diagnostic to show to user, but will save us from accidentally allowing something which we now explicitly whitelisted.